### PR TITLE
Modify rule S5722: LaYC format

### PR DESCRIPTION
**Pull request by joke1196**
## Review

A dedicated reviewer checked the rule description successfully for:

- [ ] logical errors and incorrect information
- [ ] information gaps and missing content
- [ ] text style and tone
- [ ] PR summary and labels follow [the guidelines](https://github.com/SonarSource/rspec/#to-modify-an-existing-rule)

### CODE REVIEW

File: rules/S5722/python/rule.adoc [hunk 1]
```diff
@@ -1,31 +1,25 @@
...
-The python interpreter will always call these methods with the same number of parameters. Every call to a special method will fail if it is defined with an unexpected number of parameters.
-
-
 This rule raises an issue when a special method is defined with an unexpected number of parameters.
 
+== Why is this an issue?
+
+Python allows developers to customize how code is interpreted by defining special methods (also called magic methods). For example, it is possible to override how the multiplication operator (``++a * b++``) will apply to instances of a class by defining in this class the ``++__mul__++`` and ``++__rmul__++`` methods. Whenever a multiplication operation is performed with this class, the python interpreter will call one of these methods instead of performing the default multiplication.
 
-=== Noncompliant code example
```
Comment (guillaume-dequenne-sonarsource) [inline rules/S5722/python/rule.adoc:12]: Here is missing a `How to fix it` section.

Comment (joke1196) [reply rules/S5722/python/rule.adoc:12]: I moved the code example to the how to fix section. I think it is ok like that but we could still move it back if needed as the examples are quite concise as well.

File: rules/S5722/python/rule.adoc [hunk 2]
```diff
@@ -1,33 +1,35 @@
...
+== Why is this an issue?
 
+Python allows developers to customize how code is interpreted by defining special methods (also called magic methods). For example, it is possible to override how the multiplication operator (``++a * b++``) will apply to instances of a class by defining in this class the ``++__mul__++`` and ``++__rmul__++`` methods. Whenever a multiplication operation is performed with this class, the python interpreter will call one of these methods instead of performing the default multiplication.
 
-The python interpreter will always call these methods with the same number of parameters. Every call to a special method will fail if it is defined with an unexpected number of parameters.
+Each special methods have a specific number of parameters, the Python interpreter will call these methods with those parameters. Every call to a special method will throw a ``++TypeError++`` if it is defined with an incorrect number of parameters.
```
Comment (guillaume-dequenne-sonarsource) [inline rules/S5722/python/rule.adoc:7]: I think `Each special method expects a specific number of parameters.` (with singular and full stop) sounds a bit better.
I'd also rephrase `Every call to a special method will throw a ``++TypeError++`` if it is defined with an incorrect number of parameters.`
to
`Calls to a special method will throw a ``++TypeError++`` if it is defined with an incorrect number of parameters.`

File: rules/S5722/python/rule.adoc [hunk 3]
```diff
@@ -1,33 +1,35 @@
...
 
+== How to fix it
 
-This rule raises an issue when a special method is defined with an unexpected number of parameters.
+Try using the same signature defined in the Python documentation for each special methods.
```
Comment (guillaume-dequenne-sonarsource) [inline rules/S5722/python/rule.adoc:11]: I'd say `Make sure to use the same signature as defined in the Python documentation for each special methods`.

File: rules/S5722/python/rule.adoc [hunk 4]
```diff
@@ -1,33 +1,35 @@
-== Why is this an issue?
+This rule raises an issue when a special method is defined with an unexpected number of parameters.
 
-Python developers can customize how code is interpreted by defining special methods (also called magic methods). For example, it is possible to override how the multiplication operator (``++a * b++``) will apply to instances of a class by defining in this class the ``++__mul__++`` and ``++__rmul__++`` methods. Whenever a multiplication operation is performed with this class, the python interpreter will call one of these methods instead of performing the default multiplication.
+== Why is this an issue?
 
+Python allows developers to customize how code is interpreted by defining special methods (also called magic methods). For example, it is possible to override how the multiplication operator (``++a * b++``) will apply to instances of a class by defining in this class the ``++__mul__++`` and ``++__rmul__++`` methods. Whenever a multiplication operation is performed with this class, the python interpreter will call one of these methods instead of performing the default multiplication.
```
Comment (guillaume-dequenne-sonarsource) [inline rules/S5722/python/rule.adoc:5]: Here there is an uncapitalized `python`.

**Review by guillaume-dequenne-sonarsource — CHANGES_REQUESTED** [review]
Left some suggestons.

**Review by guillaume-dequenne-sonarsource — APPROVED** [review]
LGTM.

**Review by jean-jimbo-sonarsource — APPROVED** [review]
LGTM

**Pull request merged**